server/aesdsocket.c

```diff
@@ -0,0 +1,331 @@
+    if (conn_fd == -1) {
+      if (errno == EINTR)
+        continue;
+      else {
+        syslog(LOG_ERR, "Error occured during accepting new connection = %s. Exiting...", strerror(errno));
+        exit(-1);
+      }
+    }
+    //Get IPv4 address of client
+    if (client_addr.ss_family == AF_INET) {
+      char addr[INET6_ADDRSTRLEN];
+      struct sockaddr_in * addr_in = (struct sockaddr_in * ) & client_addr;
+      inet_ntop(AF_INET, & (addr_in -> sin_addr), addr, INET_ADDRSTRLEN);
+      syslog(LOG_INFO, "Accepted connection from %s", addr);
+      printf("Accepted connection from %s\n", addr);
+    } else if (client_addr.ss_family == AF_INET6) {
+      char addr[INET6_ADDRSTRLEN];
+      struct sockaddr_in6 * addr_in6 = (struct sockaddr_in6 * ) & client_addr;
+      inet_ntop(AF_INET6, & (addr_in6 -> sin6_addr), addr, INET6_ADDRSTRLEN);
+      syslog(LOG_INFO, "Accepted connection from %s", addr);
+      printf("Accepted connection from %s\n", addr);
+    }
+
+    //Wait for data
+    int recv_ret;
+    //Start with a size, potentially incresing it if an entire packet cannot fit into it
+    recv_data = malloc(sizeof(char) * BUFFER_SIZE);
+    if(!recv_data){
+    	syslog(LOG_ERR, "Error occured during malloc = %s. Exiting...", strerror(errno));
+    	exit(-1);
+    }
+    int recv_idx = 0;
+    //How many 'BUFFER_SIZE' receive_blocks "recv_data" we have received from client till now
+    int receive_blocks = 1;
+    do {
+      recv_ret = recv(conn_fd, & recv_data[recv_idx], BUFFER_SIZE, 0);
+      if (recv_ret == -1) {
+        syslog(LOG_ERR, "Error occured while reading from socket = %s. Exiting...", strerror(errno));
+        exit(-1);
+      }
+//      printf("Received data = %s\n", &recv_data[recv_idx]);
+      recv_idx += recv_ret;
+
+      if (recv_idx != 0) {
+        if (recv_data[recv_idx - 1] == '\n') {
+
+          if (lseek(log_fd, 0, SEEK_END) == -1) {
+            syslog(LOG_ERR, "Error occured while lseek() to EOF = %s. Exiting...", strerror(errno));
+            exit(-1);
+          }
+
+          int write_byte_ret;
+          int write_len = recv_idx;
+          char * write_ptr = recv_data;
+          while (write_len != 0) {
+            write_byte_ret = write(log_fd, write_ptr, write_len);
+            
+            if (write_byte_ret == -1) {
+              if (errno == EINTR)
+                continue;
+              else {
+                syslog(LOG_ERR, "Error occured while writing to file = %s. Exiting...", strerror(errno));
+                exit(-1);
+              }
+            }
+            write_len -= write_byte_ret;
+            write_ptr += write_byte_ret;
+          }
+
+          file_size += recv_idx;
+
+          if (lseek(log_fd, 0, SEEK_SET) == -1) {
+            syslog(LOG_ERR, "Error occured while lseek() to BOF = %s. Exiting...", strerror(errno));
+            exit(-1);
+          }
+
+          char * log_read = malloc(sizeof(char) * file_size);
```

Comment:
@prvbharadwaj1731 what assumption are you making about the total size of all packets here?